### PR TITLE
Add sensor form

### PR DESCRIPTION
**Pull request by nbarlowATI**
* New frontend page to add sensor type (based on equivalent page to add location schema)
* New frontend page to add sensor (based on, but simpler than, add location page).

### CODE REVIEW

File: dtbase/webapp/app/sensors/routes.py [hunk 1]
```diff
@@ -283,3 +283,132 @@ def sensor_readings():
...
+    measure_names = request.form.getlist("measure_name[]")
+    measure_units = request.form.getlist("measure_units[]")
+    measure_datatypes = request.form.getlist("measure_datatype[]")
+    measure_existing = request.form.getlist("measure_existing[]")
+
+    # print the values
+    print("Names: ", measure_names)
+    print("Units: ", measure_units)
+    print("Datatypes: ", measure_datatypes)
+    print("Existing: ", measure_existing)
```
Comment (mastoffel) [inline dtbase/webapp/app/sensors/routes.py:323]: Can be removed.

File: dtbase/webapp/app/sensors/routes.py [hunk 2]
```diff
@@ -283,3 +283,132 @@ def sensor_readings():
...
+    sensor_types = response.json()
+    print(sensor_types)
+    return render_template("sensor_form.html", sensor_types=sensor_types)
+
+
+@login_required
+@blueprint.route("/add-sensor", methods=["POST"])
+def submit_sensor():
+    form_data = request.form
+    print(f"============={form_data}================")
```
Comment (mastoffel) [inline dtbase/webapp/app/sensors/routes.py:395]: Can be removed.

File: dtbase/webapp/app/sensors/templates/sensor_form.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <form id="sensor-form" method="POST">
+        <div class="form-group">
+            <label for="sensor_type">Sensor Type:</label>
+            <select class="form-control custom-select custom-input" id="sensor_type" name="sensor_type" onchange="updateForm()" required>
+                <option value="">-- Select a sensor type --</option>
+                {% for sensor_type in sensor_types %}
+                    <option value="{{ sensor_type.name }}">{{ sensor_type.name }}</option>
+                {% endfor %}
+            </select>
+            <label for="unique_identifier">Unique ID:</label>
```
Comment (mastoffel) [inline dtbase/webapp/app/sensors/templates/sensor_form.html:17]: Currently, it seems like we can add a second identifier with the same unique id and it shows "Sensor added successfully", but I think it's actually doing the correct thing and isn't added. So only a matter of showing the right message.

File: dtbase/webapp/app/sensors/routes.py (hunk 1 above)
Comment (mastoffel) [inline dtbase/webapp/app/sensors/routes.py:323]: Can be removed.

File: dtbase/webapp/app/sensors/routes.py (hunk 2 above)
Comment (mastoffel) [inline dtbase/webapp/app/sensors/routes.py:395]: Can be removed.

File: dtbase/webapp/app/sensors/routes.py [hunk 4]
```diff
@@ -283,3 +283,132 @@ def sensor_readings():
...
+
+    return redirect(url_for(".new_sensor_type"))
+
+
+@login_required
+@blueprint.route("/add-sensor", methods=["GET"])
+def new_sensor():
+    response = utils.backend_call("get", "/sensor/list_sensor_types")
+    sensor_types = response.json()
+    print(sensor_types)
```
Comment (mastoffel) [inline dtbase/webapp/app/sensors/routes.py:387]: Can be removed.

File: dtbase/webapp/app/sensors/routes.py (hunk 2 above)
Comment (mastoffel) [reply dtbase/webapp/app/sensors/routes.py:395]: Sorry, only the print of course :)

File: dtbase/webapp/app/sensors/routes.py (hunk 4 above)
Comment (mastoffel) [reply dtbase/webapp/app/sensors/routes.py:387]: Same here :)

File: dtbase/webapp/app/sensors/routes.py (hunk 1 above)
Comment (nbarlowATI) [reply dtbase/webapp/app/sensors/routes.py:323]: 👍

File: dtbase/webapp/app/sensors/routes.py [hunk 5]
```diff
@@ -283,3 +283,132 @@ def sensor_readings():
...
+            for idf_ex in existing_measures:
+                if idf["name"] == idf_ex["name"]:
+                    flash(
+                        f"A measure with the name '{idf['name']}' already exists.",
+                        "error",
+                    )
+                    return new_sensor_type(form_data=form_data)
+
+    try:
+        response = utils.backend_call("post", "/sensor/insert_sensor_type", form_data)
```
[the remark this answers is not in the document]
Comment (nbarlowATI) [reply dtbase/webapp/app/sensors/routes.py:369]: ahhh.... good find!   This is something I think we need to fix with the data model itself...  I made an issue about this #73

**Pull request merged**